endpoints/openrtb2/amp_auction_test.go

```diff
@@ -1934,67 +1935,68 @@ func TestSetTargeting(t *testing.T) {
 	}
 }
 
-func TestValidAmpResponseWhenRequestRejected(t *testing.T) {
-	reject := hookexecution.RejectError{
-		123,
-		hookexecution.HookID{ModuleCode: "foobar", HookImplCode: "foo"},
-		hooks.StageEntrypoint.String(),
+func TestValidAmpResponseWhenRequestStagesRejected(t *testing.T) {
+	const nbr int = 123
+	const file string = "sample-requests/amp/valid-supplementary/aliased-buyeruids.json"
+
+	var tc testCase
+	fileData, err := os.ReadFile(file)
+	assert.NoError(t, err, "Failed to read test file.")
+	if err := json.Unmarshal(fileData, &tc); err != nil {
+		t.Fatal("Failed to unmarshal test file.")
 	}
 
 	testCases := []struct {
 		description         string
-		expectedAmpResponse AmpResponse
+		file                string
+		planBuilder         hooks.ExecutionPlanBuilder
+		expectedBidResponse openrtb2.BidResponse
 		hookExecutor        hookexecution.HookStageExecutor
+		expectedAmpResponse AmpResponse
 	}{
 		{
-			"Assert correct AmpResponse when request rejected at entrypoint stage",
-			AmpResponse{Targeting: map[string]string{}},
-			rejectableHookExecutor{entrypointReject: &reject},
+			description:         "Assert correct BidResponse when request rejected at entrypoint stage",
+			file:                "sample-requests/amp/valid-supplementary/aliased-buyeruids.json",
+			planBuilder:         mockPlanBuilder{entrypointPlan: makeRejectPlan[hookstage.Entrypoint](mockRejectionHook{nbr})},
+			expectedAmpResponse: AmpResponse{Targeting: map[string]string{}},
+		},
+		{
+			description:         "Assert correct BidResponse when request rejected at raw-auction stage",
+			file:                "sample-requests/amp/valid-supplementary/aliased-buyeruids.json",
+			planBuilder:         mockPlanBuilder{rawAuctionPlan: makeRejectPlan[hookstage.RawAuctionRequest](mockRejectionHook{nbr})},
+			expectedAmpResponse: tc.ExpectedAmpResponse,
 		},
 	}
 
-	for _, test := range testCases {
-		t.Run(test.description, func(t *testing.T) {
-			actualAmpObject := analytics.AmpObject{}
-			logger := newMockLogger(&actualAmpObject, nil)
-			stored := map[string]json.RawMessage{
-				"1": json.RawMessage(validRequest(t, "site.json")),
-			}
-
-			deps := &endpointDeps{
-				fakeUUIDGenerator{},
-				&nobidExchange{},
-				mockBidderParamValidator{},
-				&mockAmpStoredReqFetcher{data: stored},
-				empty_fetcher.EmptyFetcher{},
-				empty_fetcher.EmptyFetcher{},
-				&config.Configuration{MaxRequestSize: maxSize},
-				&metricsConfig.NilMetricsEngine{},
-				logger,
-				map[string]string{},
-				false,
-				[]byte{},
-				openrtb_ext.BuildBidderMap(),
-				nil,
-				nil,
-				hardcodedResponseIPValidator{response: true},
-				empty_fetcher.EmptyFetcher{},
-				test.hookExecutor,
-			}
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			test := testCase{}
```

Comment:
I don't understand why you are creating a second test case for use with the current test case. Also you are unmarshalling `fileData` multiple times, it seems excessive.

Reply:
Most likely we overlooked to remove it after refactoring or after some merge conflict, thanks